htlcswitch/switch.go

```diff
@@ -345,15 +337,26 @@ func (s *Switch) ProcessContractResolution(msg contractcourt.ResolutionMsg) erro
 // be sent when available, or an error is encountered. If the paymentID is
 // unknown, ErrPaymentIDNotFound will be returned.
 func (s *Switch) GetPaymentResult(paymentID uint64) (<-chan *PaymentResult, error) {
-	s.pendingMutex.Lock()
-	payment, ok := s.pendingPayments[paymentID]
-	s.pendingMutex.Unlock()
+	outKey := CircuitKey{
+		ChanID: sourceHop,
+		HtlcID: paymentID,
+	}
 
-	if !ok {
-		return nil, ErrPaymentIDNotFound
+	// If the payment is not found in the circuit map, check whether a
+	// result is already available.
+	// Assumption: no one will add this payment ID other than the caller.
+	if s.circuits.LookupCircuit(outKey) == nil {
+		res, err := s.paymentResults.getPaymentResult(paymentID)
```

Comment:
`subscribePaymentResult` also returns the current state immediately, so is it necessary to have this separate block calling `getPaymentResult`?

Reply:
This is meant to handle the case where the paymentID is unknown, as we want to return an error immediately then, while `subscribePaymentResult` will wait for a result to appear.